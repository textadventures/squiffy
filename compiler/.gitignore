index.html
story.js
style.css
examples/*/jquery*.js
_site
src/jquery.min.js