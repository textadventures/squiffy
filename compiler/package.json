{
  "name": "squiffy",
  "version": "5.1.5",
  "description": "A tool for creating multiple-choice interactive stories",
  "dependencies": {
    "finalhandler": "^1.2.0",
    "glob": "^8.0.1",
    "jquery": "^3.6.0",
    "jszip": "^3.9.1",
    "marked": "^4.0.14",
    "serve-static": "^1.7.0",
    "underscore": "^1.7.0",
    "yargs": "^1.3.3"
  },
  "author": "Luis Felipe Morales <papadasoft@gmail.com>",
  "contributors": [
    "CrisisSDK <crisissdknpm@gmail.com>",
    "mrangel"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/CrisisSDK/squiffy.git"
  },
  "license": "MIT",
  "bin": "./squiffy.js",
  "preferGlobal": true
}
