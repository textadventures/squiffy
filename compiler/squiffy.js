#!/usr/bin/env node
/* jshint quotmark: single */

'use strict';

var path = require('path');
var fs = require('fs');
var compiler = require('./compiler.js');

var packageJson = JSON.parse(fs.readFileSync(path.join(__dirname, 'package.json')).toString());
var squiffyVersion = packageJson.version;

function startServer(dir, port) {
    var finalhandler = require('finalhandler');
    var http = require('http');
    var serveStatic = require('serve-static');

    var serve = serveStatic(dir, { index: ['index.html'] });

    var server = http.createServer(function(req, res){
        var done = finalhandler(req, res);
        serve(req, res, done);
    });

    server.listen(port);
}

console.log('Squiffy ' + squiffyVersion);

var argv = require('yargs')
    .usage('Compiles a Squiffy script file into HTML and JavaScript.\nFor help, see http://docs.textadventures.co.uk/squiffy/\nUsage: $0 filename.squiffy [options]')
    .demand(1)
    .alias('c', 'cdn')
    .alias('s', 'serve')
    .alias('p', 'port')
    .describe('c', 'Use CDN for jQuery')
    .describe('s', 'Start HTTP server after compiling')
    .describe('p', 'Port for HTTP server (only with --serve)')
    .describe('scriptonly', 'Only generate JavaScript file (and optionally specify a name)')
    .describe('pluginname', 'Specify the jQuery plugin name instead of .questkit (only with --scriptonly)')
    .describe('zip', 'Create zip file')
    .argv;

var options = {
    useCdn: argv.c,
    serve: argv.s,
    scriptOnly: argv.scriptonly,
    pluginName: argv.pluginname,
    zip: argv.zip,
    write: true,
};

var result = compiler.generate(argv._[0], options);

if (result && options.serve) {
    var port = argv.p || 8282;
    startServer(result, port);
    console.log('Started http://localhost:' + port + '/');
}