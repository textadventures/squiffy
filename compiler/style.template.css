a.squiffy-link
{
    text-decoration: underline;
    color: Blue;
    cursor: pointer;
}
a.squiffy-link.disabled
{
    text-decoration: inherit;
    color: inherit !important;
    cursor: inherit;
}
a.squiffy-header-link
{
    text-decoration: underline;
    color: Blue;
    cursor: pointer;
}
div#squiffy-container
{
    max-width: 700px;
    margin-left: auto;
    margin-right: auto;
    font-family: Georgia, serif;
}
div#squiffy-header
{
    font-size: 14px;
    text-align: right;
}
div#squiffy
{
    font-size: 18px;
}
hr {
    border: 0;
    height: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(255, 255, 255, 0.3);
    margin-top: 16px; margin-bottom: 16px;
}