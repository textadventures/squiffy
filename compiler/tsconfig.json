{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "module": "NodeNext",
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "skipLibCheck": true,
    "alwaysStrict": true,
    "outDir": "dist",
    "rootDir": "./src",
    "declaration": true,

    "moduleResolution": "NodeNext",
    "resolveJsonModule": true,
    "isolatedModules": true,

    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true
  },
  "include": ["src/compiler.ts"]
}
