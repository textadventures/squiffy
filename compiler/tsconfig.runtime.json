{
    "compilerOptions": {
      "target": "ES2020",
      "noImplicitAny": false,  
      "noUnusedLocals": true,
      "noUnusedParameters": true,
      "noFallthroughCasesInSwitch": true,
      "skipLibCheck": true
    },
    "include": ["src/squiffy.template.ts"]
}
  