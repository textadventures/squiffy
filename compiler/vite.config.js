import { resolve } from 'path'
import { defineConfig } from 'vite'

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/compiler.ts'),
      name: 'Squiffy',
      fileName: 'compiler',
    },
    rollupOptions: {
      // make sure to externalize deps that shouldn't be bundled
      // into your library
      external: ['jquery','marked','jszip'],
      output: {
        // Provide global variables to use in the UMD build
        // for externalized deps
        globals: {
          marked: 'marked',
        },
      },
    },
  },
})