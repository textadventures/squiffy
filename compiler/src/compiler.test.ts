import { expect, test } from 'vitest'
import * as compiler from './compiler.js';

test('some test', async () => {
    const result = await compiler.getStoryData('hello world');
    expect(result.story.start).toBe("_default");
})