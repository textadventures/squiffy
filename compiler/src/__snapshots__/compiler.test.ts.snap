// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`attributes/attributes.squiffy 1`] = `
"export const story = {};
story.id = "attributes.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "Set a start value: {{section \\"one\\" text=\\"1\\"}} or {{section \\"two\\" text=\\"2\\"}}.\\n",
        "attributes": [
            "my_string = some text"
        ]
    },
    "one": {
        "text": "Maybe you'd like it to be {{passage \\"three\\" text=\\"3\\"}} instead? Or you could just {{section \\"next\\" text=\\"continue...\\"}}\\n",
        "attributes": [
            "my_attribute = 1"
        ],
        "passages": {
            "three": {
                "text": "OK. Three it is.\\n",
                "attributes": [
                    "my_attribute = 3",
                    "chose_passage"
                ]
            }
        }
    },
    "two": {
        "text": "Maybe you'd like it to be {{passage \\"four\\" text=\\"4\\"}} instead? You could try {{passage \\"javascript\\"}}. Or you could just {{section \\"next\\" text=\\"continue...\\"}}\\n",
        "attributes": [
            "my_attribute = 2"
        ],
        "passages": {
            "javascript": {
                "text": "",
                "jsIndex": 0
            },
            "four": {
                "text": "OK. Four it is.\\n",
                "attributes": [
                    "my_attribute = 4",
                    "chose_passage"
                ]
            }
        }
    },
    "next": {
        "text": "my_attribute = {{my_attribute}}\\n\\nAlso, my_string = \\"{{my_string}}\\"\\n\\nDid you choose a passage link? {{#if chose_passage}}yes{{else}}no{{/if}} {{passage \\"unset\\"}}\\n\\nClick some more links: {{passage \\"a\\"}}, {{passage \\"b\\"}}, {{passage \\"c\\"}}, {{passage \\"d\\"}}, {{passage \\"e\\"}} and then click {{section \\"done\\"}}.\\n",
        "passages": {
            "unset": {
                "text": "Now the value is: {{#if chose_passage}}yes{{else}}no{{/if}}\\n",
                "attributes": [
                    "not chose_passage"
                ]
            },
            "a": {
                "text": "OK...\\n",
                "attributes": [
                    "count+=1"
                ]
            },
            "b": {
                "text": "OK...\\n",
                "attributes": [
                    "count+=1"
                ]
            },
            "c": {
                "text": "OK...\\n",
                "attributes": [
                    "count+=1"
                ]
            },
            "d": {
                "text": "OK...\\n",
                "attributes": [
                    "count+=1"
                ]
            },
            "e": {
                "text": "OK...\\n",
                "attributes": [
                    "count+=1"
                ]
            }
        }
    },
    "done": {
        "text": "You clicked {{count}} links. What about calling {{passage \\"dec\\"}}?\\n",
        "passages": {
            "dec": {
                "text": "Now count is {{count}}. You can also {{passage \\"increase\\"}} or {{passage \\"decrease\\"}} by values other than 1.\\n",
                "attributes": [
                    "count-=1"
                ]
            },
            "increase": {
                "text": "Now count is {{count}}.\\n",
                "attributes": [
                    "count+=10"
                ]
            },
            "decrease": {
                "text": "Now count is {{count}}.",
                "attributes": [
                    "count-=5"
                ]
            }
        }
    }
};
story.js = [
	(squiffy, get, set) => {
		alert("my_attribute " + get("my_attribute"));
	},
];"
`;

exports[`clearscreen/clearscreen.squiffy 1`] = `
"export const story = {};
story.id = "clearscreen.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "{{passage \\"one\\" text=\\"This link\\"}} doesn't clear the screen. But {{passage \\"two\\" text=\\"this link\\"}} does.\\n",
        "passages": {
            "one": {
                "text": "This text is on the same screen.\\n"
            },
            "two": {
                "clear": true,
                "text": "This text is on a new screen. This {{section \\"same page\\" text=\\"section\\"}} will be on the same screen, and this {{section \\"clear section\\" text=\\"section\\"}} clears the screen.\\n"
            }
        }
    },
    "same page": {
        "text": "This section is on the same screen.\\n"
    },
    "clear section": {
        "clear": true,
        "text": "Another new page."
    }
};
story.js = [
];"
`;

exports[`continue/continue.squiffy 1`] = `
"export const story = {};
story.id = "continue.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "Here is some text.\\n\\n{{section \\"_continue1\\" text=\\"Continue...\\"}}"
    },
    "_continue1": {
        "text": "More text.\\n\\n{{section \\"_continue2\\" text=\\"Carry on further...\\"}}"
    },
    "_continue2": {
        "text": "There's some more text now.\\n\\n{{section \\"_continue3\\" text=\\"Keep walking!\\"}}"
    },
    "_continue3": {
        "text": "Oh, even more text appeared!\\n\\n{{section \\"_continue4\\" text=\\"A little closer...\\"}}"
    },
    "_continue4": {
        "text": "Pretty scary.",
        "jsIndex": 0
    }
};
story.js = [
	(squiffy, get, set) => {
		alert("BOO!")
	},
];"
`;

exports[`helloworld/helloworld.squiffy 1`] = `
"export const story = {};
story.id = "helloworld.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "Hello, world!"
    }
};
story.js = [
];"
`;

exports[`import/test.squiffy 1`] = `
"export const story = {};
story.id = "test.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "Text from file 1.\\n\\n{{section \\"_continue1\\" text=\\"Continue...\\"}}"
    },
    "_continue1": {
        "text": "{{section \\"file2\\" text=\\"Go to file 2\\"}}\\n"
    },
    "file2": {
        "text": "Text from file 2.\\n\\n{{section \\"_continue2\\" text=\\"Continue from here...\\"}}"
    },
    "_continue2": {
        "text": "Now run the {{passage \\"test\\"}} function.\\n",
        "passages": {
            "test": {
                "text": "",
                "jsIndex": 0
            }
        }
    }
};
story.js = [
	(squiffy, get, set) => {
		test();
	},
];"
`;

exports[`last/last.squiffy 1`] = `
"export const story = {};
story.id = "last.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "Test section with some passages. {{passage \\"a\\"}}, {{passage \\"b\\"}}, {{passage \\"c\\"}}.\\n",
        "passageCount": 3,
        "passages": {
            "a": {
                "text": "Passage A.\\n"
            },
            "b": {
                "text": "Passage B.\\n"
            },
            "c": {
                "text": "Passage C.\\n"
            },
            "@last": {
                "text": "You saw all passages! Now go to the {{section \\"section2\\" text=\\"next section\\"}}.\\n"
            }
        }
    },
    "section2": {
        "text": "Second test. {{passage \\"a\\"}}, {{passage \\"b\\"}}, {{passage \\"c\\"}}.\\n",
        "passageCount": 3,
        "passages": {
            "": {
                "text": "Master passage.\\n"
            },
            "a": {
                "text": "Passage A.\\n"
            },
            "b": {
                "text": "Passage B.\\n"
            },
            "c": {
                "text": "Passage C.\\n"
            },
            "@last": {
                "text": "You saw all passages!"
            }
        }
    }
};
story.js = [
];"
`;

exports[`master/master.squiffy 1`] = `
"export const story = {};
story.id = "master.squiffy";
story.start = "intro";
story.sections = {
    "": {
        "text": "This text appears before each section in the game.\\n",
        "passages": {
            "": {
                "text": "",
                "jsIndex": 0
            }
        }
    },
    "intro": {
        "text": "This is the first section.\\n\\n{{section \\"_continue1\\" text=\\"Continue...\\"}}"
    },
    "_continue1": {
        "text": "This is the second section. {{passage \\"One\\"}}, {{passage \\"two\\"}}, {{passage \\"three\\"}}. {{section \\"Next!\\"}}\\n",
        "passages": {
            "": {
                "text": "This text appears before each passage in this section.\\n",
                "jsIndex": 1
            },
            "One": {
                "text": "1...\\n"
            },
            "two": {
                "text": "2...\\n"
            },
            "three": {
                "text": "3...\\n"
            }
        }
    },
    "Next!": {
        "text": "Final section, no master passages here. {{passage \\"See\\"}}.\\n",
        "passages": {
            "See": {
                "text": "The end..."
            }
        }
    }
};
story.js = [
	(squiffy, get, set) => {
		console.log("This JS runs for each passage in the entire game.")
	},
	(squiffy, get, set) => {
		console.log("This JS runs for each passage in the second section.");
	},
];"
`;

exports[`replace/replace.squiffy 1`] = `
"export const story = {};
story.id = "replace.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "I {{#label \\"1\\"}}walked{{/label}} to the shops and I bought {{#label \\"2\\"}}a pint of milk{{/label}}.\\n\\nBut then I thought, hang on, I {{passage \\"next\\" text=\\"don't really like milk\\" replace=\\"2=beer\\"}}.\\n",
        "passages": {
            "beer": {
                "text": "a {{passage \\"crate of beer\\"}}\\n"
            },
            "crate of beer": {
                "text": "Yes, beer.\\n"
            },
            "next": {
                "text": "And then I thought, hang on, I {{#label \\"3\\"}}{{passage \\"\\" text=\\"hate walking\\" replace=\\"1=teleported, 3=love teleporting\\"}}{{/label}}.\\n"
            },
            "love teleporting": {
                "text": "love teleporting, so much I might {{section \\"section2\\" text=\\"embark on a wild and crazy adventure\\"}}\\n"
            }
        }
    },
    "section2": {
        "text": "And so it {{#label \\"4\\"}}began...{{/label}}\\n\\n{{section \\"section3\\" text=\\"Or did it?\\"}}\\n"
    },
    "section3": {
        "text": "Oh well.\\n",
        "attributes": [
            "@replace 4=ended {{passage \\"abruptly\\"}}."
        ],
        "passages": {
            "abruptly": {
                "text": "That is all."
            }
        }
    }
};
story.js = [
];"
`;

exports[`rotate/rotate.squiffy 1`] = `
"export const story = {};
story.id = "rotate.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "Choose a number: {{rotate (array \\"one\\" \\"two\\" \\"three\\")}}\\n\\n{{section \\"_continue1\\" text=\\"OK...\\"}}"
    },
    "_continue1": {
        "text": "Now {{rotate (array \\"four\\" \\"five\\" \\"six\\") set=\\"result\\"}}\\n\\n{{section \\"_continue2\\" text=\\"OK...\\"}}"
    },
    "_continue2": {
        "text": "You chose {{result}}.\\n\\nNow for a sequence: {{sequence (array \\"one\\" \\"two\\" \\"three\\" \\"four\\" \\"five\\")}}.\\n\\n{{section \\"_continue3\\" text=\\"And another...\\"}}"
    },
    "_continue3": {
        "text": "Another sequence: {{sequence (array \\"a\\" \\"b\\" \\"c\\" \\"d\\" \\"e\\" \\"f\\" \\"g\\") set=\\"seqresult\\"}}.\\n\\n{{section \\"_continue4\\" text=\\"Finally...\\"}}"
    },
    "_continue4": {
        "text": "You chose {{seqresult}}.\\n\\nHere's a sequence that links to another section. {{sequence (array \\"Ready...\\" \\"Steady...\\" (section \\"Go!\\"))}}\\n"
    },
    "Go!": {
        "text": "That's all folks!"
    }
};
story.js = [
];"
`;

exports[`sectiontrack/sectiontrack.squiffy 1`] = `
"export const story = {};
story.id = "sectiontrack.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "Choose: {{section \\"a\\"}}, {{section \\"b\\"}}, {{section \\"c\\"}}. Or read this {{passage \\"note\\"}}.\\n",
        "passages": {
            "note": {
                "text": "This is a passage.\\n"
            }
        }
    },
    "a": {
        "text": "Now choose {{section \\"b\\"}} or {{section \\"c\\"}}.\\n"
    },
    "b": {
        "text": "Now go to {{section \\"c\\"}}.\\n"
    },
    "c": {
        "text": "{{#if (seen \\"a\\")}}You saw A.{{else}}You didn't see A.{{/if}}\\n{{#if (not (seen \\"b\\"))}}You didn't see B.{{else}}You saw B.{{/if}}\\n\\n{{#if (seen \\"note\\")}}You looked at the note.{{else}}You didn't look at the note.{{/if}}"
    }
};
story.js = [
];"
`;

exports[`start/start.squiffy 1`] = `
"export const story = {};
story.id = "start.squiffy";
story.start = "the start section";
story.sections = {
    "not the start section": {
        "text": "This is page 2, although it was defined first.\\n"
    },
    "the start section": {
        "text": "This is the start section. We also defined a section called {{section \\"not the start section\\"}}."
    }
};
story.js = [
];"
`;

exports[`test/example.squiffy 1`] = `
"export const story = {};
story.id = "example.squiffy";
story.start = "Introduction";
story.sections = {
    "Introduction": {
        "text": "# Squiffy Test\\nThis is [a website link](http://textadventures.co.uk).\\n\\nThis should be {{passage \\"passage\\" text=\\"a link to a passage\\"}}. Here's {{passage \\"passage2\\" text=\\"another one\\"}}.\\n\\nYou don't need to specify a name - for example, this {{passage \\"link\\"}} and this {{section \\"section\\"}}.\\n\\nAnd this goes to the {{section \\"section2\\" text=\\"next section\\"}}.\\n\\nThis line has links to {{section \\"section 3\\"}} and {{section \\"section four\\" text=\\"section 4\\"}}.\\n\\nThis line has links to {{passage \\"passage 3\\"}} and {{passage \\"passage four\\" text=\\"passage 4\\"}}.\\n\\nOh look - {{passage \\"it's a passage with an apostrophe\\"}}.\\n",
        "passages": {
            "passage": {
                "text": "Here's some text for the passage.\\n"
            },
            "passage2": {
                "text": "This is the text for the second passage.\\n",
                "jsIndex": 0
            },
            "passage 3": {
                "text": "Another passage.\\n"
            },
            "passage four": {
                "text": "Yet another passage.\\n"
            },
            "link": {
                "text": "Here is the text for the simple link.\\n"
            },
            "it's a passage with an apostrophe": {
                "text": "Nothing to worry about.\\n"
            }
        }
    },
    "section2": {
        "text": "This is the next section, with passages {{passage \\"a\\"}} and {{passage \\"b\\"}}.\\n",
        "jsIndex": 1,
        "passages": {
            "a": {
                "text": "Just a.\\n"
            },
            "b": {
                "text": "Just b.\\n"
            }
        }
    },
    "section": {
        "text": "This is another section, with passages {{passage \\"a\\"}} and {{passage \\"b\\"}}.\\n",
        "passages": {
            "a": {
                "text": "Just a in a different section.\\n"
            },
            "b": {
                "text": "Just b in a different section.\\n"
            }
        }
    },
    "section 3": {
        "text": "Another section is here, with passages {{passage \\"a\\"}} and {{passage \\"b\\"}}.\\n",
        "passages": {
            "a": {
                "text": "Just a in section3.\\n"
            },
            "b": {
                "text": "Just b in section3.\\n"
            }
        }
    },
    "section four": {
        "text": "Some other section, with passages {{passage \\"a\\"}} and {{passage \\"b\\"}}.\\n",
        "passages": {
            "a": {
                "text": "Just a in section four.\\n"
            },
            "b": {
                "text": "Just b in section four."
            }
        }
    }
};
story.js = [
	(squiffy, get, set) => {
		alert ("This passage has some JavaScript");
	},
	(squiffy, get, set) => {
		alert ("This section has some JavaScript");
	},
];"
`;

exports[`textprocessor/textprocessor.squiffy 1`] = `
"export const story = {};
story.id = "textprocessor.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "It's another {{random (array \\"Monday\\" \\"Tuesday\\" \\"Wednesday\\" \\"Thursday\\" \\"Friday\\" \\"Saturday\\" \\"Sunday\\")}} in the city.\\n\\nThe weather is {{random (array \\"sunny\\" \\"cloudy\\" \\"rainy\\" \\"windy\\" \\"snowy\\") set=\\"weather\\"}}.\\n\\nChoose a gender:\\n\\n* {{section \\"start\\" text=\\"Male\\" set=\\"gender=male\\"}}\\n* {{section \\"start\\" text=\\"Female\\" set=\\"gender=female\\"}}\\n"
    },
    "start": {
        "text": "You chose {{gender}}.\\n\\n{{#if (eq gender \\"male\\")}}\\nYou are a man.\\n{{else}}\\nYou are a woman.\\n{{/if}}\\n\\nIt's still {{weather}}.\\n\\n{{passage \\"Next\\"}}\\n",
        "passages": {
            "Next": {
                "text": "This is a test. {{embed \\"passage\\"}} {{embed \\"section\\"}}\\n"
            },
            "passage": {
                "text": "Text from passage.\\n"
            }
        }
    },
    "section": {
        "text": "Text from section."
    }
};
story.js = [
];"
`;

exports[`transitions/transitions.squiffy 1`] = `
"export const story = {};
story.id = "transitions.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "You are standing in a room. You can see a {{passage \\"hammer\\"}}.\\n",
        "passages": {
            "hammer": {
                "text": "It's nice and weighty. Perhaps you want to {{section \\"hit\\" text=\\"hit yourself in the face with it\\"}}?\\n"
            }
        }
    },
    "hit": {
        "text": "Bang.\\n",
        "jsIndex": 0
    },
    "wake up": {
        "clear": true,
        "text": "Slowly your eyes open.\\n\\nThat was a rather stupid thing to do."
    }
};
story.js = [
	(squiffy, get, set) => {
		squiffy.ui.transition(function(){
			// Create a blackout <div>...
			$("<div/>", {
				id: "blackout"
			}).css({
				position: "fixed",
				top: 0,
				left: 0,
				width: "100%",
				height: "100%",
				background: "black",
				display: "none"
			}).appendTo("body");
			// Fade it in over 6 seconds...
			$("#blackout").fadeIn({
				duration: 6000,
				easing: "easeInQuad",
			});
			// After fading it in...
			setTimeout(function () {
				// Move to the next section...
				squiffy.story.go("wake up");
				// Fade out the blackout and remove the <div>...
				$("#blackout").fadeOut({
					duration: 6000,
					easing: "easeInQuad",
					complete: function() {
						$(this).remove();
					}
				});
			}, 9000);
		});
		
	},
];"
`;

exports[`turncount/turncount.squiffy 1`] = `
"export const story = {};
story.id = "turncount.squiffy";
story.start = "_default";
story.sections = {
    "_default": {
        "text": "After clicking two links in this section, an additional passage will be displayed.\\n\\n{{passage \\"One\\"}}, {{passage \\"two\\"}}, {{passage \\"three\\"}}, {{passage \\"four\\"}}.\\n",
        "passages": {
            "One": {
                "text": "First passage.\\n"
            },
            "two": {
                "text": "Second passage.\\n"
            },
            "three": {
                "text": "Third passage.\\n"
            },
            "four": {
                "text": "Fourth passage.\\n"
            },
            "@2": {
                "text": "This is some extra text to appear after the second click. The {{section \\"next section\\"}} is also good.\\n"
            }
        }
    },
    "next section": {
        "text": "In this section, you can only click one link, and then you'll be taken to the final section.\\n\\n{{passage \\"One\\"}}, {{passage \\"two\\"}}, {{passage \\"three\\"}}, {{passage \\"four\\"}}.\\n",
        "passages": {
            "One": {
                "text": "First passage.\\n"
            },
            "two": {
                "text": "Second passage.\\n"
            },
            "three": {
                "text": "Third passage.\\n"
            },
            "four": {
                "text": "Fourth passage.\\n"
            },
            "@1": {
                "text": "",
                "jsIndex": 0
            }
        }
    },
    "final section": {
        "text": "**THE END**"
    }
};
story.js = [
	(squiffy, get, set) => {
		squiffy.story.go("final section");
	},
];"
`;

exports[`warnings/warnings.squiffy 1`] = `
[
  "WARNING: warnings.squiffy line 1: In section '_default' there is a link to a section called [[badsection2]], which doesn't exist",
  "WARNING: warnings.squiffy line 1: In section '_default' there is a link to a section called [[badsection]], which doesn't exist",
  "WARNING: warnings.squiffy line 9: In section 'section' there is a link to a passage called [badpassage2], which doesn't exist",
  "WARNING: warnings.squiffy line 9: In section 'section' there is a link to a passage called [badpassage], which doesn't exist",
  "WARNING: warnings.squiffy line 19: In section 'section', passage 'passage' there is a link to a passage called [non-existent passage], which doesn't exist",
]
`;

exports[`warnings/warnings2.squiffy 1`] = `
[
  "WARNING: warnings2.squiffy line 1: In section 'test' there is a link to a passage called [doesn't exist], which doesn't exist",
]
`;
